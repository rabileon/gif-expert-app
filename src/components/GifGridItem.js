import React from 'react';

export const GifGridItem = ({ id, title, url }) => {
  console.log({ id, title, url });
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title}></img>
    </div>
  );
};

/* 
1. Enzyme
2. Ensyme to json
3. Debe mostrar el componente correctamente
 * shallow
 * wrapper
 * wrapper .toMatchSnapshot()
 */
