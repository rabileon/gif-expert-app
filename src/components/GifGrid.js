import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>
      {loading && (
        <p className="loading animate__animated flash">Cargando ...</p>
      )}
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
GifGrid.protoTypes = {
  category: PropTypes.string.isRequired,
};
