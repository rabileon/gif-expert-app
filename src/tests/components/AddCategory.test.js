import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory/>', () => {
  const setCategory = () => {};
  const wrapper = shallow(<AddCategory setCategories={setCategory} />);
  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de camiar la caja de texto', () => {
    const input = wrapper.find('input');
    const value = 'Hola Mundo';

    input.simulate('change', { target: { value } });

    // expect (wrapper.find('p').text().trim()).toBe(value );
  });
});
