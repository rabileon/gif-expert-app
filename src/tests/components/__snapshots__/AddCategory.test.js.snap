// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory/> debe de mostrarse correctamente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Busca un gif, escribe cualquier palabra y presiona enter"
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
